core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -44,7 +45,9 @@ object RequestChannel extends Logging {
     byteBuffer
   }
 
-  case class Request(processor: Int, connectionId: String, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
+  case class Session(principal: Principal, host: String)
```

Comment:
Oh, nice. :) :+1: Yeah... maybe ``ConnectionState`` or ``ConnectionInfo`` could be a better name. Just a suggestion.